webhooks/data-import-cron.go

```diff
@@ -0,0 +1,217 @@
+		if len(accessModes) != 1 {
+			return fmt.Errorf("required value: exactly 1 access mode is required")
+		}
+		if err := validateAccessMode(accessModes[0]); err != nil {
+			return err
+		}
+		if spec.PVC.DataSource != nil || spec.PVC.DataSourceRef != nil {
+			return fmt.Errorf("external population is incompatible with DataImportCrons")
+		}
+	} else if spec.Storage != nil {
+		for _, accessMode := range spec.Storage.AccessModes {
+			if err := validateAccessMode(accessMode); err != nil {
+				return err
+			}
+		}
+		if spec.Storage.DataSource != nil || spec.Storage.DataSourceRef != nil {
+			return fmt.Errorf("external population is incompatible with DataImportCrons")
+		}
+	}
+
+	if err := validateNumberOfSources(spec.Source); err != nil {
+		return err
+	}
+
+	if registry := spec.Source.Registry; registry != nil {
+		if err := validateDataVolumeSourceRegistry(registry); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateStorageSize(spec *cdiv1beta1.DataVolumeSpec) error {
+	var name string
+	var resourceRequests core.ResourceList
+
+	if spec.PVC != nil {
+		resourceRequests = spec.PVC.Resources.Requests
+		name = "PVC"
+	} else if spec.Storage != nil {
+		resourceRequests = spec.Storage.Resources.Requests
+		name = "Storage"
+	}
+
+	storageSize, ok := resourceRequests["storage"]
+	if !ok {
+		return fmt.Errorf("%s size is missing", name)
+	}
+	if storageSize.Value() <= 0 {
+		return fmt.Errorf("%s size can't be equal or less than zero", name)
+	}
+	return nil
+}
+
+func validateStorageClassName(spec *cdiv1beta1.DataVolumeSpec) error {
+	var sc *string
+
+	if spec.PVC != nil {
+		sc = spec.PVC.StorageClassName
+	} else if spec.Storage != nil {
+		sc = spec.Storage.StorageClassName
+	}
+
+	if sc == nil || *sc == "" {
+		return nil
+	}
+
+	if scErrors := validation.IsDNS1123Subdomain(*sc); len(scErrors) > 0 {
+		return fmt.Errorf("invalid storage class name: %q", scErrors)
+	}
+	return nil
+}
+
+func validateAccessMode(accessMode core.PersistentVolumeAccessMode) error {
+	switch accessMode {
+	case core.ReadWriteOnce:
+	case core.ReadOnlyMany:
+	case core.ReadWriteMany:
+	case core.ReadWriteOncePod:
+	default:
+		return fmt.Errorf(`unsupported value: "%s": supported values: "ReadOnlyMany", "ReadWriteMany", "ReadWriteOnce", "ReadWriteOncePod"`, string(accessMode))
+	}
+	return nil
+}
+
+func validateNumberOfSources(source *cdiv1beta1.DataVolumeSource) error {
+	numberOfSources := 0
+	s := reflect.ValueOf(source).Elem()
+	for i := 0; i < s.NumField(); i++ {
+		if !reflect.ValueOf(s.Field(i).Interface()).IsNil() {
+			numberOfSources++
+		}
+	}
+	if numberOfSources == 0 {
+		return fmt.Errorf("missing DataVolume source")
+	}
+	if numberOfSources > 1 {
+		return fmt.Errorf("multiple DataVolume sources")
+	}
+	return nil
+}
+
+func validateDataVolumeSourceRegistry(sourceRegistry *cdiv1beta1.DataVolumeSourceRegistry) error {
+	sourceURL := sourceRegistry.URL
+	sourceIS := sourceRegistry.ImageStream
```

Comment:
Up to you but I'd inline most of these vars

Reply:
I'd keep them this way. It would be too verbose if inlined.